plugins/common/tls/config.go

```diff
@@ -206,23 +210,62 @@ func (c *ServerConfig) TLSConfig() (*tls.Config, error) {
 func makeCertPool(certFiles []string) (*x509.CertPool, error) {
 	pool := x509.NewCertPool()
 	for _, certFile := range certFiles {
-		pem, err := os.ReadFile(certFile)
+		cert, err := os.ReadFile(certFile)
 		if err != nil {
 			return nil, fmt.Errorf("could not read certificate %q: %w", certFile, err)
 		}
-		if !pool.AppendCertsFromPEM(pem) {
+		if !pool.AppendCertsFromPEM(cert) {
 			return nil, fmt.Errorf("could not parse any PEM certificates %q: %w", certFile, err)
 		}
 	}
 	return pool, nil
 }
 
-func loadCertificate(config *tls.Config, certFile, keyFile string) error {
-	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+func loadCertificate(config *tls.Config, certFile, keyFile, privateKeyPassphrase string) error {
+	var (
+		cert      tls.Certificate
+		certBytes []byte
+		keyBytes  []byte
+		err       error
+	)
+	certBytes, err = os.ReadFile(certFile)
 	if err != nil {
-		return fmt.Errorf("could not load keypair %s:%s: %w", certFile, keyFile, err)
+		return err
 	}
 
+	keyBytes, err = os.ReadFile(keyFile)
+	if err != nil {
+		return err
+	}
+
+	keyPEMBlock, _ := pem.Decode(keyBytes)
```

Comment:
Please check the returned error!

Reply:
`pem.Decode` does not really return an error. It returns a block which is a `*pem.Block` and a byte slice which is the rest of the bytes after the PEM block. If no PEM block is found, the whole of the input is returned as second return value and the PEM block is `nil`. 

Here, we rely on the value of the PEM block and treat the `nil` value as an error.